backend/packages/Upgrade/src/api/repositories/AnalyticsRepository.ts

```diff
@@ -485,22 +456,60 @@ export class AnalyticsRepository {
       .limit(take)
       .where('monitoredDecisionPoint.experimentId = :experimentId', { experimentId });
     
-    const [individualEnrollmentQueryResult, monitoredDecisionPointQueryResult] = await Promise.all([
+    const [individualEnrollmentQueryResults, monitoredDecisionPointQueryResults] = await Promise.all([
```

Comment:
I'm not sure we can be guaranteed that the data from these two queries will match up for every iteration of the loop. If there are multiple monitored decision point entries for a given experiment and student, and only one enrollment, for instance, we could read 50 records from `montiored_decision_point` before we get the the entries for that student, even though we'll have the entry in `individual_enrollment`. That said, we're considering removing the loop anyway.

Reply:
Removed skip take from `montiored_decision_point` query in the last commit so that we guarantee that we take all MDPlogs for the users from `individual_enrollment` query. If you want, you can carry forward from here to remove skip take throughout.